crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -508,33 +545,82 @@ impl KitsuneP2pHandler for Space {
+                            Ok(response) => {
+                                tracing::debug!(msg = "Peer found via MDNS", ?response);
+                                // Decode response
+                                let remote_agent_vec = base64::decode_config(
+                                    &response.service_name[..],
+                                    base64::URL_SAFE_NO_PAD,
+                                )
+                                .expect("Agent base64 decode failed");
+                                let remote_agent = Arc::new(KitsuneAgent(remote_agent_vec));
+                                //println!("(MDNS) - Peer found via MDNS: {:?})", *remote_agent);
+                                let maybe_agent_info_signed =
+                                    kitsune_p2p_types::codec::rmp_decode(&mut &*response.buffer);
+                                if let Err(e) = maybe_agent_info_signed {
+                                    tracing::error!(msg = "Failed to decode peer from MDNS", ?e);
+                                    continue;
+                                }
+                                let remote_agent_info_signed = maybe_agent_info_signed.unwrap();
+                                //println!("(MDNS) - Found agent_info_signed: {:?})", remote_agent_info_signed);
+                                // Add to local storage
+                                let _result = evt_sender
+                                    .put_agent_info_signed(PutAgentInfoSignedEvt {
+                                        space: space.clone(),
+                                        agent: remote_agent,
+                                        agent_info_signed: remote_agent_info_signed,
+                                    })
+                                    .await;
+                            }
+                            Err(e) => {
+                                tracing::error!(msg = "Failed to get peers from MDNS", ?e);
+                            }
+                        }
                     }
+                });
+            }
+            NetworkType::QuicBootstrap => {
+                let bootstrap_service = self.config.bootstrap_service.clone();
+                if let Some(bootstrap_service) = bootstrap_service {
+                    tokio::task::spawn(async move {
+                        const START_DELAY: std::time::Duration = std::time::Duration::from_secs(1);
+                        const MAX_DELAY: std::time::Duration =
+                            std::time::Duration::from_secs(60 * 60);
+                        let mut delay_len = START_DELAY;
+
+                        loop {
+                            tokio::time::delay_for(delay_len).await;
+                            if delay_len <= MAX_DELAY {
+                                delay_len *= 2;
+                            }
+
+                            if let Err(e) = super::discover::add_5_or_less_non_local_agents(
```

Comment:
```suggestion
                            // TODO - this will make redundant requests to bootstrap server if multiple local agents have joined the same space.
                            if let Err(e) = super::discover::add_5_or_less_non_local_agents(
```